pkg/agent/entrypoint.go

```diff
@@ -79,18 +75,31 @@ func (r MainRunner) Run(logger *zap.Logger, ctx context.Context) error {
 		}
 	}
 
-	logger.Info("Entering main loop")
-	for {
-		event, err := vmEventQueue.Wait(ctx)
-		if err != nil {
-			if ctx.Err() != nil {
-				// treat context canceled as a "normal" exit (because it is)
-				return nil
-			}
+	mc, err := billing.NewMetricsCollector(ctx, logger, &r.Config.Billing)
+	if err != nil {
+		return fmt.Errorf("error creating billing metrics collector: %w", err)
+	}
 
-			logger.Error("vmEventQueue returned error", zap.Error(err))
-			return err
+	tg := taskgroup.NewGroup(logger, taskgroup.WithParentContext(ctx))
+	tg.Go("billing-metrics", func(logger *zap.Logger) error {
```

Comment:
I _think_ actually just changing this to `"billing"` would resolve my [comments from earlier](https://github.com/neondatabase/autoscaling/pull/922#discussion_r1631329536) -- keeping the logger names the same.

Reading through it again (please check me on this!), AFAICT we would still end up with

`.billing.collect` as the main collection thread:

https://github.com/neondatabase/autoscaling/blob/954e2c64ed9ab3545f3327f702e2dff428bfb065/pkg/agent/billing/billing.go#L168-L169

and `.billing.send-$CLIENT` as the sending threads

https://github.com/neondatabase/autoscaling/blob/954e2c64ed9ab3545f3327f702e2dff428bfb065/pkg/agent/billing/billing.go#L165

---

It'd probably be good to move the sender threads to task group as well (especially so that we make sure we're actually waiting for them to finish before exiting!), but I'm happy to leave that for a follow-up.